rero_ils/modules/items/api/issue.py

```diff
@@ -47,3 +52,195 @@ def issue_is_regular(self):
+            .sort({'_created': {'order': 'asc'}}) \
+            .source(['pid']).scan()
+        for hit in results:
+            yield hit.pid
+
+    @classmethod
+    def receive_next_late_expected_issues(cls, dbcommit=False, reindex=False):
+        """Receive the next late expected issue for all holdings.
+
+        :param reindex: reindex record by record.
+        :param dbcommit: commit record to database.
+
+        :return a count of created issues.
+        """
+        from ...holdings.api import Holding
+        created_issues = 0
+        pids = cls.get_late_serial_holdings_pids()
+        for pid in pids:
+            holding = Holding.get_record_by_pid(pid)
+            issue = holding.receive_regular_issue(
+                dbcommit=dbcommit, reindex=reindex)
+            issue['issue']['status'] = ItemIssueStatus.LATE
+            issue = issue.update(issue, dbcommit=dbcommit, reindex=reindex)
+            created_issues += 1
+        return created_issues
+
+    @classmethod
+    def _process_issue_claim(
+            cls, issue, claims_count, claims_days, modified_issues,
+            dbcommit=False, reindex=False):
+        """Process issue for a new claim.
+
+        :param issue: the issue item record.
+        :param claims_days: claims days parameter to compare with issue age.
+        :param modified_issues: counter to count modified issues.
+        :param reindex: reindex the records.
+        :param dbcommit: commit record to database.
+        :param claims_count: current claim count for this issue.
+
+        :return a count of modified issues.
+        """
+        days = cls._get_issue_status_age(issue)
+        if days >= claims_days:
+            cls._update_issue_status_claims_count(
+                issue, claims_count,
+                dbcommit=dbcommit, reindex=reindex)
+            modified_issues += 1
+        return modified_issues
+
+    @classmethod
+    def _get_issue_status_age(cls, issue):
+        """Return the age of an issue.
+
+        It compares the issue status date with the current timestamp
+
+        :param issue: the issue item record.
+        :return the issue status age in days.
+        """
+        issue_status_date = ciso8601.parse_datetime(
+            issue.issue_status_date)
+        return (datetime.now(timezone.utc) - issue_status_date).days
+
+    @classmethod
+    def _update_issue_status_claims_count(
+            cls, issue, claims_count, dbcommit=False,
+            reindex=False):
+        """Update issue status and claims count.
+
+        It compares the issue status date with the current timestamp
+
+        :param issue: the issue item record.
+        :param claims_count: the current claims_count from the issue.
+        :param reindex: reindex the records.
+        :param dbcommit: commit record to database.
+        """
+        issue['issue']['status'] = ItemIssueStatus.CLAIMED
+        issue['issue']['claims_count'] = issue.claims_count + 1
```

Comment:
Maybe we can use a `setter` here:

```
    @property
    def claims_count(self):
        """Shortcut for the issue claims_count."""
        return self.get('issue', {}).get('claims_count', 0)

    @claims_count.setter
    def claims_count(self, claims_count):
        self.setdefault('issue', {}).setdefault('claims_count', 0)
        self['issue']['claims_count'] = claims_count
```
Then we can write `issue.claims_count += 1` ???